### PR TITLE
[Automated] Merge release into main

### PR DESCRIPTION
**Pull request by ProKil**
Automated Pull Request.

This PR also pins mypy version to avoid its latest bug. https://github.com/python/mypy/pull/17543

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ ruff = "*"
 pytest = "*"
 pytest-cov = "*"
 pytest-asyncio = "*"
-mypy = "^1.8.0"
+mypy = ">=1.8.0,<1.11.0"
```
Comment (XuhuiZhou) [inline pyproject.toml:52]: I thought 1.11.0 is the buggy version?

File: .github/workflows/merge-release-to-main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Merge Release Into Main
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'release'
+
+jobs:
```
Comment (XuhuiZhou) [inline .github/workflows/merge-release-to-main.yml:9]: what does this do? I thought the main should be the most updated branch? Or it's more like the release branch simply changes the version name?

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ProKil) [reply pyproject.toml:52]: so <1.11.0 is the right range.

File: .github/workflows/merge-release-to-main.yml (hunk 2 above)
Comment (XuhuiZhou) [reply .github/workflows/merge-release-to-main.yml:9]: Cool, thanks for the explanation! I think this is worthwhile to add to the contribution doc, which I will do that.

**Review by XuhuiZhou — APPROVED** [review]

**Pull request merged**